src/main/java/org/mdbenefits/app/submission/actions/HandleApplicationSigned.java

```diff
@@ -19,15 +20,21 @@ public class HandleApplicationSigned implements Action {
 
     private final SubmissionRepositoryService submissionRepositoryService;
 
+    private final JdbcTemplate jdbcTemplate;
+
     public HandleApplicationSigned(MessageSource messageSource, MailgunEmailClient mailgunEmailClient,
-            SubmissionRepositoryService submissionRepositoryService) {
+            SubmissionRepositoryService submissionRepositoryService, JdbcTemplate jdbcTemplate) {
         this.messageSource = messageSource;
         this.mailgunEmailClient = mailgunEmailClient;
         this.submissionRepositoryService = submissionRepositoryService;
+        this.jdbcTemplate = jdbcTemplate;
     }
 
     @Override
     public void run(Submission submission) {
+        String confirmationNumber = jdbcTemplate.queryForObject("select 'M-' || nextval('confirmation_sequence')", String.class);
+        submission.getInputData().put("confirmationNumber", confirmationNumber);
+        log.info("Using confirmationNumber: {}", confirmationNumber);
         sendEmailToApplicant(submission);
```

Comment:
This does not work as expected. We need to explicitly save the submission again here, else the confirmation number will be gone and not actually stored in the data.

It is not saved automatically as the submission object is not considered "managed" anymore (I think). 

What I did was run this code w/o an email recipient.  The confirmation number was lost. 

When I ran with a recipient address, the confirmation number is save correctly, but only because the section that sends the email will call save explicitly.